### PR TITLE
[16.0][MIG] account_invoice_supplier_ref_unique: Migration to 16.0

### PR DESCRIPTION
**Pull request by SMaciasOSI**
/ocabot migration account_invoice_supplier_ref_unique

### CODE REVIEW

**Review by rousseldenis — APPROVED** [review]
Code review.

IMHO, such constraints should be able to be bypassed (checkbox or...). Maybe in ROADMAP

**Comment by SMaciasOSI** [comment]
@max3903 @rousseldenis Can this PR be merged?

**Comment by rousseldenis** [comment]
@max3903 Could you fix pylint by replacing format by %(placeholder)s ?

**Comment by max3903** [comment]
> @max3903 Could you fix pylint by replacing format by %(placeholder)s ?

@SMaciasOSI Let me know if you need help with this...

File: account_invoice_supplier_ref_unique/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                            "=ilike",
+                            rec.supplier_invoice_number,
+                        ),
+                        ("id", "!=", rec.id),
+                    ],
+                    limit=1,
+                )
+                if same_supplier_inv_num:
+                    raise ValidationError(
+                        _("%(placeholder)s").format(
```
Comment (rousseldenis) [inline account_invoice_supplier_ref_unique/models/account_move.py]: @SMaciasOSI This is the correct one:

```suggestion
                         _(
                            "The invoice/refund with supplier invoice number %(number)s "
                            "already exists in Odoo under the number %(same)s "
                            "for supplier %(supplier)s.",
                            dict(
                                number=same_supplier_inv_num.supplier_invoice_number,
                                same=same_supplier_inv_num.name or "-",
                                supplier=same_supplier_inv_num.partner_id.display_name)
                        )
```

**Review by max3903 — APPROVED** [review]

File: account_invoice_supplier_ref_unique/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                    ],
+                    limit=1,
+                )
+                if same_supplier_inv_num:
+                    raise ValidationError(
+                        _(
+                            "The invoice/refund with supplier invoice number %(number)s "
+                            "already exists in Odoo under the number %(same)s "
+                            "for supplier %(supplier)s.",
+                            dict(
```
Comment (rousseldenis) [inline account_invoice_supplier_ref_unique/models/account_move.py]: ooops, in fact it's just args, so remove 'dict('

**Pull request merged**